parser/README.md

```diff
@@ -2,17 +2,32 @@
 
 This crate contains the parser for AirScript.
 
-The purpose of the parser is to parse the constraints written in human friendly AirScript language into an AST (Abstract Syntax Tree).
+The purpose of the parser is to parse the constraints written in the human-friendly AirScript language into an Abstract Syntax Tree.
 
-## Generating AST
+## Generating the Abstract Syntax Tree (AST)
 
-The parser uses [Logos](https://github.com/maciejhirsz/logos/) to write a custom lexer which is then fed into the parser generated using [LALRPOP](https://github.com/lalrpop/lalrpop/).
+The parser uses [Logos](https://github.com/maciejhirsz/logos/) to generate a custom lexer, which is then fed into the parser generated by [LALRPOP](https://github.com/lalrpop/lalrpop/).
 
-To create an AST from a given AIR constraints module, you just need to pass your source to the public `parse` function, which will return the AST or an `Error` of type `ScanError` or `ParseError`. `parse` will first tokenize the source using the lexer, then map the tokens to tokens accepted by the parser which are of type `(usize, Token, usize)`. Invalid tokens will be stored as `ScanError`. Finally, `parse` feeds the tokens to the parser to generate the corresponding AST (or `ParseError`).
+To create an AST from a given AirScript module, pass your source to the public `parse` function, which will return the AST or an `Error` of type `ScanError` or `ParseError`.
 
-For example:
+`parse` will first tokenize the source using the lexer, then map the resulting tokens to new tokens accepted by the parser, which are of type `(usize, Token, usize)`. Each invalid token will be stored as `ScanError`. Finally, if no `ScanError` occurred, `parse` feeds the tokens to the parser to generate a Result with the corresponding AST (or `ParseError`).
```

Comment:
Minor nit: Similar comment as above. It might be better to write "The `parse` function will".